### ISSUE TITLE
iOS App?

### ISSUE THREAD

**Issue by Smarthome-Creator** [SEPIA-Framework/sepia-docs#41]
Hallo,

eigentlich nur eine ganz kurze Frage. Gibt es Sepia eigentlich für iOS? Ich habe zwar in deiner Dokumentation was davon gelesen, kann Sepia aber nicht im App Store finden.

Gruß

**Comment by sepia-assistant** [comment]
Hi,

ja und nein 😅 .

Es gibt die App, aber man muss sie selber mit den vorhandenen Skripten bauen und dann in Xcode laden um sie aufs Handy zu kopieren.
Das hat 2 Gründe. Zum einen wollte ich nicht die 90€ Apple Developer Gebühren pro Jahr zahlen und den zusätzlichen Release Overhead ertragen, bevor es sich nicht wirklich lohnt und zweitens habe ich selber kein aktuelles Gerät mehr zum Testen, seitdem mein altes iPhone 6 auf iOS 12 festhängt :-( (der MacOS Simulator ist nur begrenzt nützlich).

Die App ist im Prinzip voll funktionsfähig, bis auf die Siri Spracherkennung. Apple hatte ein Swift Update erzungen was die Library kaputt gemacht hat und ich kam noch nicht dazu das neu zu machen. Der SEPIA STT Server funktioniert aber.

Falls genug Interesse besteht könnte ich den Teil wiederbeleben :-)

**Comment by Smarthome-Creator** [comment]
Ja verstehe ich. Da die App ja eigentlich "nur" auf die HTML Seite des Servers zugreift, könnte es dann vielleicht über den iOS Safari laufen?

**Comment by fquirin** [comment]
Ja, manche Sachen funktionieren ganz gut als PWA. Leider ist Apple da auch sehr konservativ und sowas wie Notifications für Timer wird man auf absehbare Zeit wohl nicht hinbekommen, aber die Spracheingabe über den SEPIA STT Server müsste klappen. Ich werde das noch mal testen mit dem neusten iOS :+1:

**Comment by Smarthome-Creator** [comment]
Ich habe es mal mit verschiedenen Safari Versionen versucht und ja man kann sich mit dem mit Sepia verbinden, ist allerdings nur halb so komfortabel am Tablet wie die App. Ganz generell wäre die App natürlich toll, weil so ein iPad einfach besser an der Wand aussieht als so ein Androide. Wenn allerdings immer nur das neuste iOS unterstützt wird, dann lohnt es sich natürlich auch nicht wirklich. Ein Tutorial wie man dass mit Xcode zusammenfummelt und auf´s Gerät überträgt wäre sicher auch ganz gut.

Aber erstmal ganz entspannt, drängelt jetzt nicht.

**Comment by cap-blackbeard** [comment]
Ich würde auf meinem iPhone gerne den SEPIA benutzen.
Meine Erfahrungen mit Safari:

- web-Client kann ich per https laden
- bei jeder Anmeldung fragt mich Safari, ob die Seite auf das Mikrofon zugreifen kann, ww und STT funktionieren dann aber
- STT wird doppelt verstanden
- TTS wird blockiert trotz https
- scheinbar wird GPS auch geblockt, SEPIA fehlen die Informationen zu meinem Standort
- hatte versucht die Seite zum Home-Bildschirm hinzuzufügen. Dann wir aber irgendwie das Mikrofon ebenfalls unterdrückt.

Wollen wir das mit der App nochmal versuchen?

**Comment by fquirin** [comment]
Hi,

ja, Apple und Safari sind echt super nervig :-|. Die guten Neuigkeiten sind, seit dem iOS Update von dieser Woche auf 15.4. ist zumindest scheinbar der "STT wird doppelt verstanden" Bug der Apple WebSpeechApi endlich behoben. Es gab auch einen Bug nachdem das Mikrofon nicht mehr richtig funktioniert hat wenn man Musik abgespielt hatte. Das scheint jetzt auch zu gehen, zumindest auf iOS, Mac OS Safari ist immer noch seltsam, da kann man aber wenigstens "richtige" Browser installieren =).

Ich habe meinen alten Mac Mini mal auf den neusten Stand gebracht und das aktuelle Xcode installiert, vielleicht versuche ich noch mal, ob ich die App bauen kann, aber wahrscheinlich sind einige Cordova Plugins kaputt.
Kennst du dich mit Xcode aus? ^^

Das grundsätzliche Problem mit der iOS App ist dass man viel Zeit investieren muss um die am laufen zu halen und Apple die Sache auch noch kompliziert macht weil man ständig den Developer Account bezahlen muss für 90€ im Jahr oder so (was ich nicht mache ^^).

**Comment by cap-blackbeard** [comment]
Nö. Mit Xcode kenne ich mich nicht aus. Habe mal etwas im Netz gestöbert. Da ich keinen Mac besitze müsste ich wohl per VMbox Mac OS installieren und dann ggf. noch mit son nem Tool (USB Network Gate) die USB-Verbindung herstellen?
Ich habe ich bisher auch noch nicht damit beschäftigt. Was benötigt man, um eine nicht durch Apple zugelassene App ohne AppStore auf dem iPhone zu installieren? Xcode?

> habe ich selber kein aktuelles Gerät mehr zum Testen, seitdem mein altes iPhone 6 auf iOS 12 festhängt :-( 

Grundsätzlich bin ich eben auch kein Apple-Jünger. Aber mit nem iPhone11 könnte ich hier ggf. helfen


> zumindest scheinbar der "STT wird doppelt verstanden" Bug der Apple WebSpeechApi endlich behoben

Jap. Habe soeben das Update gemacht. STT funktioniert. Kein doppeltes Erkennen mehr da.

**Comment by fquirin** [comment]
> Da ich keinen Mac besitze müsste ich wohl per VMbox Mac OS installieren und dann ggf. noch mit son nem Tool (USB Network Gate) die USB-Verbindung herstellen?

Lieber nicht! Damit habe ich nur schlechte Erfahrung :see_no_evil: 

> Grundsätzlich bin ich eben auch kein Apple-Jünger. Aber mit nem iPhone11 könnte ich hier ggf. helfen

Danke, ich hatte zwischenzeitlich auch mal ein iPhone 7 ergattert für ne kleine Mark :-)

> Jap. Habe soeben das Update gemacht. STT funktioniert. Kein doppeltes Erkennen mehr da.

Cool, immerhin etwas ^^. Wenn man die App auf den Homescreen legt, sperrt Apple den Zugriff allerdings wieder :-(. Ich war irgendwann mal so naiv zu glauben Web-Apps (PWAs) hätten eine realistische Chance auf iOS, aber damit verdient Apple ja kein Geld :-|.

> Ich habe ich bisher auch noch nicht damit beschäftigt. Was benötigt man, um eine nicht durch Apple zugelassene App ohne AppStore auf dem iPhone zu installieren? Xcode?

Grundsätzlich ja und Xcode läuft auch nur auf Mac :angry: (wobei wohl auch auf iPads demnächst ^^). Es gibt noch Test-Flight von Apple, da muss aber glaube ich mindestens der App Ersteller einen bezahlten Account haben und ich glaube die App läuft dann alle 2 Wochen ab. Alles in allem mehr als nervig :grimacing:

**Comment by cap-blackbeard** [comment]
Hmm... also kommen wir hier vorerst nicht weiter.

**Comment by fquirin** [comment]
Ich fürchte ja, teste aber gleich mal wie der aktuelle Stand ist.

**Comment by fquirin** [comment]
@cap-blackbeard 
Ich habe mal eine Bestandsaufnahme gemacht ... und direkt diverse Bugs gefixt :laughing: 
Die gute Nachricht: Es läuft eigentlich alles bis auf die Siri-Spracherkennung ("native" STT) und YouTube Media-Player :grinning: 
Die "Schlechte": Apple benötigt immer noch einen 99 Dollar Developer Account bevor man User als Beta-Tester einladen kann :unamused: 
Vielleicht starte ich irgendwann mal eine Spenden-Aktion für eine SEPIA iOS public Beta ;-)
